datasets/sst/sst.py

```diff
@@ -0,0 +1,198 @@
+
+import csv
+import os
+
+import datasets
+
+
+_CITATION = """\
+@inproceedings{socher-etal-2013-recursive,
+    title = "Recursive Deep Models for Semantic Compositionality Over a Sentiment Treebank",
+    author = "Socher, Richard and Perelygin, Alex and Wu, Jean and
+      Chuang, Jason and Manning, Christopher D. and Ng, Andrew and Potts, Christopher",
+    booktitle = "Proceedings of the 2013 Conference on Empirical Methods in Natural Language Processing",
+    month = oct,
+    year = "2013",
+    address = "Seattle, Washington, USA",
+    publisher = "Association for Computational Linguistics",
+    url = "https://www.aclweb.org/anthology/D13-1170",
+    pages = "1631--1642",
+}
+"""
+
+_DESCRIPTION = """\
+The Stanford Sentiment Treebank, the first corpus with fully labeled parse trees that allows for a
+complete analysis of the compositional effects of sentiment in language.
+"""
+
+_HOMEPAGE = "https://nlp.stanford.edu/sentiment/"
+
+_LICENSE = ""
+
+_URL = "https://www.dropbox.com/s/9j9dc55hs28wrye/stanfordSentimentTreebank.zip?dl=1"
+
+
+class Sst(datasets.GeneratorBasedBuilder):
+    """The Stanford Sentiment Treebank"""
+
+    VERSION = datasets.Version("1.0.0")
+
+    BUILDER_CONFIGS = [
+        datasets.BuilderConfig(
+            name="default",
+            version=VERSION,
+            description="Sentences and relative parse trees annotated with sentiment labels.",
+        ),
+        datasets.BuilderConfig(
+            name="dictionary",
+            version=VERSION,
+            description="List of all possible sub-sentences (phrases) with their sentiment label.",
+        ),
+        datasets.BuilderConfig(
+            name="ptb", version=VERSION, description="Penn Treebank-formatted trees with labelled sub-sentences."
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = "default"
+
+    def _info(self):
+
+        if self.config.name == "default":
+            features = datasets.Features(
+                {
+                    "sentence": datasets.Value("string"),
+                    "label": datasets.Value("float"),
+                    "tokens": datasets.Value("string"),
+                    "tree": datasets.Value("string"),
+                }
+            )
+        elif self.config.name == "dictionary":
+            features = datasets.Features({"phrase": datasets.Value("string"), "label": datasets.Value("float")})
+        else:
+            features = datasets.Features(
+                {
+                    "ptb_tree": datasets.Value("string"),
+                }
+            )
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            supervised_keys=None,
+            homepage=_HOMEPAGE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        data_dir = dl_manager.download_and_extract(_URL)
+
+        file_paths = {}
+        for split_index in range(0, 4):
```

Comment:
Why do you need to copy the dictionary 4 times ?

Reply:
3 for train, validation and test in the `default` configuration. The last one is for the only split in the `dictionary` configuration. I am not sure if there's a more clever way to do this